File: Javascript_sol/Arrays_hashing/Leetcode_229.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var majorityElement = function(nums) {
  let tt = parseInt(nums.length/3);
  let hmap = {};
  let arr = [];

  nums.forEach((num, ind) => {
      if (hmap[num]) {
          hmap[num] += 1
      } else {hmap[num] = 1};
      if (hmap[num]>tt) {
          arr.push(num)
      }
  });
  arr = Array.from(new Set(arr))
  return arr
};

File: Python_sol/Arrays_hashing/Leetcode_229.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        hmap = {}
        arr = []
        tt = len(nums)//3

        # Time complexity O(2N) Space complexity O(2N)
        for i in nums:
            if hmap.get(i,0):
                hmap[i] += 1
            else:
                hmap[i] = 1

            if hmap[i] > tt:
                    arr.append(i)
        return set(arr)